src/optimizer/rule/ordered_aggregate_optimizer.cpp

```diff
@@ -24,6 +30,90 @@ unique_ptr<Expression> OrderedAggregateOptimizer::Apply(LogicalOperator &op, vec
 		changes_made = true;
 		return nullptr;
 	}
+
+	auto &context = rewriter.context;
+	if (context.config.enable_optimizer) {
```

Comment:
I think this condition is not necessary since the `OrderedAggregateOptimizer` is already an optimizer which shouldn't run when optimizations are disabled - IIRC we had this in the code before because we were doing optimizations in the binding phase

Reply:
Oh, hm, is this optimization now happening twice (also in `src/function/aggregate/sorted_aggregate_function.cpp`)?

I would say it should happen preferably only in the optimizer indeed as it is an optimization - and the binding only does the actual binding of the function.

Reply:
Turn out there is code that uses this without going through the optimiser (`test/sql/aggregate/distinct/issue8505.test`).